internal/plugin/server.go

```diff
@@ -326,69 +326,71 @@ func (plugin *NvidiaDevicePlugin) Allocate(ctx context.Context, reqs *pluginapi.
+		response.Devices = append(response.Devices, plugin.apiDeviceSpecs(*plugin.config.Flags.NvidiaDriverRoot, requestIds)...)
 	}
 	if *plugin.config.Flags.GDSEnabled {
 		response.Envs["NVIDIA_GDS"] = "enabled"
 	}
 	if *plugin.config.Flags.MOFEDEnabled {
 		response.Envs["NVIDIA_MOFED"] = "enabled"
 	}
+	return response, nil
+}
+
+func (plugin NvidiaDevicePlugin) updateAllocatResponseForMPS(response *pluginapi.ContainerAllocateResponse, deviceIDs ...string) {
 	// TODO: We should generate a CDI specification for MPS
-	if plugin.config.Sharing.SharingStrategy() == spec.SharingStrategyMPS {
-		if response.Envs == nil {
-			response.Envs = make(map[string]string)
-		}
-		pipeDir := filepath.Join("/mps", string(plugin.rm.Resource()), "pipe")
-		response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = pipeDir
-		response.Mounts = append(response.Mounts,
-			&pluginapi.Mount{
-				ContainerPath: pipeDir,
-				HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", pipeDir),
-			},
-		)
-		logDir := filepath.Join("/mps", string(plugin.rm.Resource()), "log")
-		response.Envs["CUDA_MPS_LOG_DIRECTORY"] = logDir
-		response.Mounts = append(response.Mounts,
-			&pluginapi.Mount{
-				ContainerPath: logDir,
-				HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", logDir),
-			},
-		)
-		response.Mounts = append(response.Mounts,
-			&pluginapi.Mount{
-				ContainerPath: "/dev/shm",
-				HostPath:      "/var/lib/kubelet/device-plugins/mps/shm",
-			},
-		)
-	}
-
-	return &response, nil
+	pipeDir := filepath.Join("/mps", string(plugin.rm.Resource()), "pipe")
+	response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = pipeDir
+	response.Mounts = append(response.Mounts,
+		&pluginapi.Mount{
+			ContainerPath: pipeDir,
+			HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", pipeDir),
+		},
+	)
+	logDir := filepath.Join("/mps", string(plugin.rm.Resource()), "log")
+	response.Envs["CUDA_MPS_LOG_DIRECTORY"] = logDir
+	response.Mounts = append(response.Mounts,
+		&pluginapi.Mount{
+			ContainerPath: logDir,
+			HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", logDir),
+		},
+	)
+	response.Mounts = append(response.Mounts,
+		&pluginapi.Mount{
+			ContainerPath: "/dev/shm",
+			HostPath:      "/var/lib/kubelet/device-plugins/mps/shm",
+		},
+	)
 }
 
-// getAllocateResponseForCDI returns the allocate response for the specified device IDs.
+// updateAllocateResponseForCDI updates the specified response for the given device IDs.
 // This response contains the annotations required to trigger CDI injection in the container engine or nvidia-container-runtime.
-func (plugin *NvidiaDevicePlugin) getAllocateResponseForCDI(responseID string, deviceIDs []string) (pluginapi.ContainerAllocateResponse, error) {
-	response := pluginapi.ContainerAllocateResponse{}
-
-	if !plugin.cdiEnabled {
-		return response, nil
-	}
-
+// TODO: We should generate a CDI specification for MPS
```

Comment:
Does this imply (at the moment) that MPS is **only** supported without CDI enabled?

Reply:
I have removed the TODOs related to using CDI for MPS injection.